### PR TITLE
Allow to override build date with SOURCE_DATE_EPOCH

### PR DESCRIPTION
**Pull request by bmwiedemann**
It changes the following:
- checks for `SOURCE_DATE_EPOCH` env and uses it as `build_date`

I followed the checklist:
- [ ] I added unittests
- [X] I ran `make check` without errors
- [ ] I tested the overall application
- [X] I added an appropriate item to the changelog

Allow to override build date with `SOURCE_DATE_EPOCH`
to make builds reproducible.
See https://reproducible-builds.org/ for why this is good
and https://reproducible-builds.org/specs/source-date-epoch/
for the definition of this variable.

This patch was done while working on [reproducible builds for openSUSE](https://en.opensuse.org/openSUSE:Reproducible_Builds).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+* support reproducible builds ([#2202](https://github.com/extrawurst/gitui/pull/2202))
```
Comment (extrawurst) [inline CHANGELOG.md:10]: * lets be more specific, cause this PR only adds a way to pin the build time stamp
* please look at other contributors and how they are named in the changelog

**Comment by extrawurst** [comment]
thanks for the contribution, just a very small nitpick, otherwise good to go

File: CHANGELOG.md (hunk 1 above)
Comment (bmwiedemann) [reply CHANGELOG.md]: I updated the line. Should be better now.

**Comment by bmwiedemann** [comment]
Are more changes needed?

**Comment by extrawurst** [comment]
Can you fix the CI issues?

**Comment by extrawurst** [comment]
@bmwiedemann it really is nothing more than running cargo fmt so that it formats the `build.rs` file you editted

**Pull request merged**